### PR TITLE
[ RNN ] Add Multi-Layer RNN Unit Tests 

### PR DESCRIPTION
**Pull request by jijoongmoon**
This commit includes,
 . Multi-Layered RNN Unit Tests
 . Some Fixes to run

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/layers/rnn.cpp [hunk 1]
```diff
@@ -291,7 +286,7 @@ void RNNLayer::calcGradient() {
       djdb_h.add_i(dh);
       djdw_x.add_i(xs.dot(dh, true, false));
```
Comment (kparichay) [inline nntrainer/layers/rnn.cpp:287]:
```
xs.dot(dh, djdw_x, true, false, 1.0);
hs_prev.dot(dh, djdw_h, true, false, 1.0);
```

Please consider using above

**Review by zhoonit — APPROVED** [review]
LGTM

File: nntrainer/layers/rnn.cpp (hunk 1 above)
Comment (jijoongmoon) [reply nntrainer/layers/rnn.cpp]: done.

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**